OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -750,6 +763,45 @@ static bool Team(S server, Connection conn, Session.Client client, string s)
 			return true;
 		}
 
+		static bool ToggleSpawn(S server, Connection conn, Session.Client client, string s)
+		{
+			if (!client.IsAdmin)
+			{
+				server.SendOrderTo(conn, "Message", "Only admins can disable spawn points.");
+				return true;
+			}
+
+			var spawnPoint = Exts.ParseIntegerInvariant(s);
+			if (spawnPoint == 0)
+				return true;
+
+			var existingClient = server.LobbyInfo.Clients.FirstOrDefault(cc => cc.SpawnPoint == spawnPoint);
+
+			if (!server.LobbyInfo.DisabledSpawns.Contains(spawnPoint))
+			{
+				// check we have enough to toggle
+				if (server.Map.SpawnPoints.Length - server.LobbyInfo.Clients.Count(c => !c.IsObserver) - server.LobbyInfo.DisabledSpawns.Count < (existingClient != null ? 1 : 0))
+				{
+					server.SendOrderTo(conn, "Message", "Unable to disable as there would be insufficient spawn locations for all players.");
```

Comment:
Agree, perhaps:

* Unable to disable spawn location, players need a start location.
* All players will require a start location so spawn is still open.

Reply:
There is another edge case to worry about here: the admin can disable the maximum number of allowed spawns, but there is nothing stopping a spectator from switching into an empty player slot, now causing there to be less free spawns than players. That player can ready up, but nothing will happen (not even a warning to the chat about insufficient free slots).

IMO trying to deal with that edge case and coming up with sensible error messages for the different cases is more complexity than it is worth. It would be better if we could instead allow the admin to disable as many spawns as they want, and have a single clear error message explaining that the game can't start until the situation is resolved.

Reply:
This is mostly the behaviour now - you can disable as many as you like, the only problem piece missing is the lack of notification why the game can't start currently.